cranelift/frontend/src/switch.rs

```diff
@@ -454,29 +460,28 @@ block3:
 block0:
     v0 = iconst.i8 0
     v1 = icmp_imm uge v0, 7  ; v0 = 0
-    brnz v1, block9
-    jump block8
+    brif v1, block9, block8
 
 block9:
     v2 = icmp_imm.i8 uge v0, 10  ; v0 = 0
-    brnz v2, block10
-    jump block11
+    brif v2, block11, block10
 
-block11:
+block10:
     v3 = icmp_imm.i8 eq v0, 7  ; v0 = 0
-    brnz v3, block4
-    jump block0
+    brif v3, block4, block0
 
 block8:
     v4 = icmp_imm.i8 eq v0, 5  ; v0 = 0
-    brnz v4, block3
-    jump block12
+    brif v4, block3, block12
 
 block12:
+    jump block13
+
+block13:
```

Comment:
Similarly, this unnecessary intermediate block is introduced as an effect of the changes to `build_search_branches`, and is the subject of the TODO on line 451.